### PR TITLE
Story/ccls 1814 get add linked cases

### PR DESCRIPTION
**Pull request by PhilDigitalJustice**
includes ccls-1815

### CODE REVIEW

File: caab-service/src/main/java/uk/gov/laa/ccms/caab/api/service/ApplicationService.java [hunk 1]
```diff
@@ -69,6 +71,100 @@ public ApplicationDetail getApplication(final Long applicationId) {
...
+  /**
+   * Creates and associates a new linked case with a specified application.
+   * If the application is not found, a CaabApiException is thrown.
+   *
+   * @param id The unique identifier of the application.
+   * @param linkedCase The LinkedCase object containing the details of the case to be linked.
+   * @throws CaabApiException If the application with the specified ID is not found.
+   */
+  @Transactional
+  public void createLinkedCaseForApplication(final Long id, final LinkedCase linkedCase) {
```
Comment (porritta) [inline caab-service/src/main/java/uk/gov/laa/ccms/caab/api/service/ApplicationService.java:99]: Change param name to `applicationId` for ease of understanding

File: caab-service/src/main/java/uk/gov/laa/ccms/caab/api/service/ApplicationService.java [hunk 2]
```diff
@@ -69,6 +71,100 @@ public ApplicationDetail getApplication(final Long applicationId) {
...
+   * Removes a linked case from the specified application.
+   * If either the application or the linked case is not found, a CaabApiException is thrown.
+   *
+   * @param id The unique identifier of the application.
+   * @param linkedCaseId The unique identifier of the linked case to be removed.
+   * @throws CaabApiException If the application or the linked case with the specified IDs are
+   *        not found.
+   */
+  @Transactional
+  public void removeLinkedCaseFromApplication(final Long id, final Long linkedCaseId) {
```
Comment (porritta) [inline caab-service/src/main/java/uk/gov/laa/ccms/caab/api/service/ApplicationService.java:121]: Change to applicationId

File: caab-service/src/main/java/uk/gov/laa/ccms/caab/api/service/ApplicationService.java [hunk 3]
```diff
@@ -69,6 +71,100 @@ public ApplicationDetail getApplication(final Long applicationId) {
...
+   *
+   * @param id The unique identifier of the application.
+   * @param linkedCaseId The unique identifier of the linked case to be updated.
+   * @param linkedCaseModel The LinkedCase object containing the details of the case to be updated.
+   * @throws CaabApiException If the application or the linked case with the specified IDs are not
+   *        found.
+   */
+  @Transactional
+  public void updateLinkedCaseForApplication(
+      final Long id, final Long linkedCaseId,
```
Comment (porritta) [inline caab-service/src/main/java/uk/gov/laa/ccms/caab/api/service/ApplicationService.java:149]: Change to applicationId

**Review by porritta — COMMENTED** [review]
review complete

File: caab-service/src/main/java/uk/gov/laa/ccms/caab/api/service/ApplicationService.java (hunk 1 above)
Comment (PhilDigitalJustice) [reply caab-service/src/main/java/uk/gov/laa/ccms/caab/api/service/ApplicationService.java:130]: done

File: caab-service/src/main/java/uk/gov/laa/ccms/caab/api/service/ApplicationService.java (hunk 2 above)
Comment (PhilDigitalJustice) [reply caab-service/src/main/java/uk/gov/laa/ccms/caab/api/service/ApplicationService.java:152]: done

**Review by porritta — APPROVED** [review]
LGTM

**Pull request merged**